Car.cpp

```diff
@@ -1,7 +1,41 @@
 #include "Car.hpp"
 #include <iostream>
 
-void Car::turnLeft()      { std::cout << __FUNCTION__ << std::endl; }
-void Car::turnRight()     { std::cout << __FUNCTION__ << std::endl; }
-void Car::brake()         { std::cout << __FUNCTION__ << std::endl; }
-void Car::accelerate(int) { std::cout << __FUNCTION__ << std::endl; }
+void Car::turnLeft()
+{
+    if(velocity!=0)
+        std::cout << __FUNCTION__ << std::endl;
+    else
+        std::cout << "You must move to turn!" << std::endl;
+}
+void Car::turnRight()
+{
+    if(velocity!=0)
+        std::cout << __FUNCTION__ << std::endl;
+    else
+        std::cout << "You must move to turn!" << std::endl;
+}
+void Car::accelerate(int speed)
+{ 
+    std::cout << __FUNCTION__ << std::endl;
+    if (velocity += speed < -30)
+        Car::velocity = -30;

```

Comment:
skad jest to "-30"?

Reply:
z głowy, założyłem, że samochód nie powinien jechać do tylu więcej niż 30. wolałbyś jakaś zmienna jak w przypadku Vmax?

Reply:
Na pewno. Magiczne wartości, które rozumie tylko autor kodu są złą praktyką :) Jeśli by to było odpowiednio nazwane to ok.